autorest/codegen/models/utils.py

```diff
@@ -4,5 +4,13 @@
 # license information.
 # --------------------------------------------------------------------------
 import re
+from enum import Enum
 
 JSON_REGEXP = re.compile(r'^(application|text)/(.+\+)?json$')
+
+AAD_TOKEN = "AADToken"
+AZURE_KEY = "AzureKey"
+class Credential(Enum):
```

Comment:
> if you really want to do a credential enum, can you change it to a credential model instead? It's easier to read and more pythonic. You can either create a new file in the `models` folder or add it to an existing one.

I move the enum definition to `CredentialInfo`, is that ok?

Reply:
> additionally, if we're going to move people to using security instead, it might be a good idea to keep the credential policy information that currently exists on the code model as a property on `Credential`, then switch code model to have a `credential` property instead. That way, the code path we want users to use (security), has much easier code, bc you don't have to switch from credential to credential policy

I think it is better to refactor it as you described when we are ready to support two kinds of auth at the same time.

Reply:
@msyyc i still think it's better to create it as a model, then a dict with a bunch of enums. It's harder to read the current way, and it doesn't flow with the other definitions we have.

I still think it's better to refactor now, bc the way it is currently in the PR, we have duplicated info. This is because we have `credential_policy` on the client, and also a credential policy enum in `CredentialInfo`, and we always want to avoid duplication. So I think it's better to move the highlighted code [here](https://github.com/Azure/autorest.python/blob/autorestv3/autorest/codegen/models/code_model.py#L214-L225) onto a credential model, and then switch code model to just have a new `credential` property. That way it's clear where everything is

Reply:
Fine. Let me refactor it now.